frame/fees/src/lib.rs

```diff
@@ -532,111 +525,28 @@ pub mod pallet {
-        return Ok(0);
-      }
-
-      let currency_wanted = (usdt_amount * order_book_price)
-        .checked_div(&FixedU128::from(
-          10_u128.pow(Asset::Tether.exponent() as u32),
-        ))
-        .ok_or(Error::<T>::BalanceOverflow)?;
-
-      Ok(
-        currency_wanted
-          .checked_mul(&FixedU128::from(10_u128.pow(asset_from.exponent() as u32)))
-          .ok_or(Error::<T>::InvalidUsdtValue)?
-          .into_inner()
-          .checked_div(FixedU128::DIV)
-          .ok_or(Error::<T>::BalanceOverflow)?,
-      )
-    }
-
-    // Based on the value provided by Oracle, try to convert the asset balance to USDT balance
-    pub(crate) fn try_get_usdt_value(
-      currency_id: CurrencyId,
-      amount: FixedU128,
-    ) -> Result<Balance, DispatchError> {
-      let asset_from: Asset = currency_id
-        .try_into()
-        .map_err(|_| Error::<T>::InvalidAsset)?;
-
-      let order_book_price = if currency_id == CurrencyId::Wrapped(USDT) {
-        FixedU128::from(1)
-      } else {
-        Self::order_book_price(currency_id, CurrencyId::Wrapped(USDT))
-      };
-
-      if order_book_price.is_zero() {
-        return Ok(0);
-      }
-
-      let currency_wanted = amount
-        .checked_div(&order_book_price)
-        .ok_or(Error::<T>::InvalidUsdtValue)?
-        .checked_div(&FixedU128::from(10_u128.pow(asset_from.exponent() as u32)))
-        .ok_or(Error::<T>::BalanceOverflow)?;
-
-      Ok(
-        currency_wanted
-          .saturating_mul(FixedU128::from(
-            10_u128.pow(Asset::Tether.exponent() as u32),
-          ))
-          .into_inner()
-          .checked_div(FixedU128::DIV)
-          .ok_or(Error::<T>::InvalidUsdtValue)?,
-      )
-    }
-
     // Based on the value provided by Oracle, try to convert the asset balance to TDFY balance
-    pub(crate) fn try_get_tide_value(
+    pub(crate) fn try_get_tdfy_value(
       currency_id: CurrencyId,
-      amount: FixedU128,
+      amount: Balance,
     ) -> Result<Balance, DispatchError> {
       let asset_from: Asset = currency_id
         .try_into()
         .map_err(|_| Error::<T>::InvalidAsset)?;
 
-      let order_book_price = if currency_id == CurrencyId::Tdfy {
-        FixedU128::from(1)
-      } else {
-        Self::order_book_price(currency_id, CurrencyId::Tdfy)
+      let order_book_price = match currency_id {
+        CurrencyId::Tdfy => FixedU128::from(1),
+        CurrencyId::Wrapped(asset_id) => Self::wrapped_asset_value(asset_id),
       };
 
       if order_book_price.is_zero() {
         return Ok(0);
       }
 
-      let currency_wanted = amount
-        .checked_div(&order_book_price)
-        .ok_or(Error::<T>::InvalidTdfyValue)?
-        .checked_div(&FixedU128::from(10_u128.pow(asset_from.exponent() as u32)))
-        .ok_or(Error::<T>::BalanceOverflow)?;
+      let amount = FixedU128::saturating_from_rational(amount, asset_from.saturating_mul(1))
+        .saturating_mul(order_book_price);
```

Comment:
Here we need to divide the amount by order_book_price, or multiple by (1/order_book_price).

Reply:
we are getting a rational fixedu128 from the amount sent from oracle so we don't need to do any multiple_by here